git_machete/github.py

```diff
@@ -90,73 +97,131 @@ def _token_from_hub() -> Optional[str]:
     return None
 
 
-def _token_from_env() -> Optional[str]:
+def __token_from_env() -> Optional[str]:
     return os.environ.get(GITHUB_TOKEN_ENV_VAR)
 
 
 def github_token() -> Optional[str]:
-    return _token_from_env() or _token_from_gh() or _token_from_hub()
+    return __token_from_env() or __token_from_gh() or __token_from_hub()
 
 
-def fire_github_api_get_request(url: str, token: Optional[str]) -> Any:
+def __fire_github_api_request(method: str, url: str, token: Optional[str], request_body: Optional[Dict[str, Any]] = None) -> Any:
     headers: Dict[str, str] = {
         'Content-type': 'application/json',
-        'User-Agent': 'git-machete'
+        'User-Agent': 'git-machete',
+        'Accept': 'application/vnd.github.v3+json'
     }
     if token:
         headers['Authorization'] = 'Bearer ' + token
 
-    host = 'api.' + GITHUB_DOMAIN
-    conn: HTTPSConnection = HTTPSConnection(host)
+    host = 'https://api.' + GITHUB_DOMAIN
+    http_request = Request(host + url, headers=headers, data=json.dumps(request_body).encode(), method=method.upper())
 
     try:
-        conn.request('GET', url, body=None, headers=headers)
-
-        response: HTTPResponse = conn.getresponse()
-        body: Any = json.loads(response.read().decode())
-
-        if 200 <= response.status < 300:
-            return body
+        with urlopen(http_request) as response:
+            parsed_response_body: Any = json.loads(response.read().decode())
+            return parsed_response_body
+    except HTTPError as err:
+        if err.code == 422:
```

Comment:
Nit: does urrlib provide some constant instead, like `UNPROCESSABLE_ENTITY`? I mean, 422 is relatively obscure code compared to e.g. 400 or 404, I can imagine it might get confused one day once this `if` grows

Reply:
(oh okay, didn't see the one above)

Reply:
Ideally if it were imported from `urllib` to use just one library or everything... or is `urllib` a wrapper over `http` anyway?

Reply:
unfortunatelly it's a wrapper around http.client

Reply:
Okay, then using a constant from `http` is justified